nsv13/code/modules/overmap/weapons/mines.dm

```diff
@@ -0,0 +1,99 @@
+	layer = ABOVE_MOB_LAYER
+	animate_movement = NO_STEPS
+	max_integrity = 300
+	integrity_failure = 100
+	var/datum/star_system/current_system
+	var/faction = "syndicate" //evil mines
+	var/damage = 100
+	var/damage_type = BRUTE
+	var/damage_flag = "overmap_heavy"
+	alpha = 110 //They're supposed to be sneaky, their main advantage is being cloaked
+
+/obj/structure/space_mine/Initialize(mapload, var/new_faction, var/datum/star_system/system, var/list/coordinates)
+	. = ..()
+	if(system)
+		current_system = system
+	else if(!current_system) //Someone is probably spawning us on the overmap, so we assume it's next to the main ship
+		current_system = SSstar_system.find_main_overmap().current_system
+	current_system.system_contents |= src
+	if(new_faction)
+		faction = new_faction
+	update_icon()
+	if(coordinates)
+		x = coordinates["x"]
+		y = coordinates["y"]
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/space_mine/Destroy(force)
+	current_system?.contents_positions.Remove(src)
+	current_system?.system_contents.Remove(src)
+	RemoveElement(/datum/element/connect_loc)
+	. = ..()
+
+/// This makes us not drift like normal objects in space do
+/obj/structure/space_mine/Process_Spacemove(movement_dir = 0)
+	return 1
+
+/obj/structure/space_mine/proc/on_entered(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+
+	if(istype(AM, /obj/item/projectile))
+		var/obj/item/projectile/P = AM
+		if(P.faction != faction)
+			P.Impact(src)
+
+	if(istype(AM,/obj/structure/overmap))
+		var/obj/structure/overmap/OM = AM
+		if(OM.faction != faction)
+			mine_explode(OM)
+
+/obj/structure/space_mine/update_icon(updates)
+	. = ..()
+	if(faction)
+		icon_state = "mine_[faction]"
+	else
+		icon_state = "mine_unaligned"
+		faction = "unaligned"
+
+/obj/structure/space_mine/obj_break(damage_flag)
+	if(prob(80))
+		obj_destruction()
+	else //Whoops, IFF broke!
+		faction = "unaligned"
+
+/obj/structure/space_mine/obj_destruction(damage_flag)
+	mine_explode() //Why you mine explode? To the woods with you
+	. = ..()
+
+/obj/structure/space_mine/proc/mine_explode(obj/structure/overmap/OM)
+	var/armour_penetration = 0
+	if(OM) //You just flew into a mine
+		armour_penetration = 20 //It's blowing up right next to you, this is what it was designed for
+		if(OM.use_armour_quadrants)
+			OM.take_quadrant_hit(OM.run_obj_armor(damage, damage_type, damage_flag, null, armour_penetration), OM.quadrant_impact(src))
+		else
+			OM.take_damage(damage, damage_type, damage_flag, FALSE, TRUE)
+		if(OM.linked_areas) //Hope nothing precious was in that room.
+			var/area/A = pick(OM.linked_areas)
+			var/turf/T = pick(get_area_turfs(A))
+			new /obj/effect/temp_visual/explosion_telegraph(T, damage)
+	else
+		for(var/obj/structure/overmap/O in orange(2)) //You're in range! Keep in mind this affects *all* ships, explosions don't discriminate between friend and foe
```

Comment:
```suggestion
		for(var/obj/structure/overmap/O in orange(2, src)) //You're in range! Keep in mind this affects *all* ships, explosions don't discriminate between friend and foe
```

Orange center arg defaults to usr which is probably one of the worst inbuilt vars that exists in byond.

Reply:
Pff okay lol